beacon_node/lighthouse_network/src/discovery/mod.rs

```diff
@@ -177,6 +177,12 @@ pub struct Discovery<TSpec: EthSpec> {
     /// always false.
     pub started: bool,
 
+    /// This keeps track of whether an external UDP port change should also indicate an internal
+    /// TCP port change. As we cannot detect our external TCP port, we assume that the external UDP
+    /// port is also our external TCP port. This assumption only holds if the user has not
+    /// explicitly set their ENR TCP port via the CLI config.
```

Comment:
An user can also set both the `--discovery-port` and `-port` to use different values. If UDP and TCP are bound to different ports I assume they will have different external ports too, don't you think? Also, we need to start taking into account tcp6